docs/adr/0002-release-and-branching-strategy.md

```diff
@@ -0,0 +1,208 @@
+  * Good, because it covers the needs of most common development and release patterns.
+  * Good, because it isolates tested code from the `develop` branch by intentionally keeping `main` clear of unfinished or untested work. (Would resolve issues like [#325](https://github.com/trussworks/react-uswds/issues/325)).
+  * Good, because it offers the option of using built-in tooling via the git-flow wrapper for git.
+  * Good, because it scales well for a large number of collaborators.
+  * Good, because it is foundational to a variety of more tailored branching strategies.
+  * Bad, because it is slightly more intricate than the [Default](#default) strategy.
+  * Bad, because it involves an extra branch (`develop`) meaning more PRs and merges are necessary to manage code and releases.
+  * Bad, because the generic Gitflow strategy does not cover *all* development or release needs, and is generally designed for extension, so it is a common trap to feel the need to contort in order to remain within the bounds of generic Gitflow guidelines.
+  * Bad, because it does not provide out-of-the-box guidance for navigating development dependencies, and may not address the [problem statement](#context-and-problem-statement) without changes to the release process as well.
+
+#### [Modified Gitflow](https://docs.google.com/document/d/1FGW1u3W6T1M-HSw8Jw3xcjM_8HdnOsV5rdbYphPuLxY/edit#)
+* Characteristics
+  * An extension of the [default branching strategy](#default) via tailored additions of [Gitflow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow) concepts to match development and release needs of the `react-uswds` project.
+  * `main` is the primary develop branch.
+  * Features are PRed to `main`.
+  * Bugfixes are PRed to `main`.
+  * Additional develop branches are the primary extension of the default branching strategy
+  * Releases take place via short-lived release branches based off `main`, tagged on `main`.
+    * As noted below, this strategy also supports custom releases from the additional develop branches, if desired.
+* Pros and Cons
+  * Good, because it scales extremely well to *both* custom developer workflows and custom release strategies
+    * During times of low developer availability or periods without development dependencies the branching strategy can be scaled to be effectively identical to the [Default](#default) strategy.
+    * Its flexibility supports multiple release strategies, ideal for future changes to that process.
+  * Good, because it provides a clear process for navigating development dependencies and unblocking corresponding work.
+  * Good, because it retains the familiar process of PRing off of and to a base develop branch.
+  * Bad because, while based on well documented and widely understood concepts, this custom branching strategy would require additional documentation to make clear for new contributors.
```

Comment:
Let's mitigate this by adding documentation for:
- How to start a new USWDS version branch (this should probably start with the version upgrade issue)
- Working on USWDS features that are ahead of the current supported version
- Knowing when & how to release any breaking changes (this is primarily a judgement call depending on the severity of the change)

...and also by making sure we're using the USWDS version milestones and `breaking` labels consistently. Maybe we add a label to indicate issues related to the current version as an easy way for contributors to find things to work on?